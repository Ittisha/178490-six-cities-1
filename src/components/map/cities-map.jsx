import * as React from 'react';
import PropTypes from 'prop-types';
import leaflet from 'leaflet';

const ZOOM = 12;

export class CitiesMap extends React.PureComponent {
  render() {
    return (
      <div id="map" style={{height: `100%`}} />
    );
  }

  componentDidMount() {
    this._createMap();
    setTimeout(() => this._map.invalidateSize(true), 400);
  }

  componentDidUpdate() {
    this._updateMarkers();
  }

  componentWillUnmount() {
    this._map.remove();
    this._map = null;
  }

  _createMap() {
    const {initCoords} = this.props;
    this._map = leaflet.map(`map`, {
      center: initCoords,
      zoom: ZOOM,
      zoomControl: false,
      marker: true
    });

    leaflet
      .tileLayer(`https://{s}.basemaps.cartocdn.com/rastertiles/voyager/{z}/{x}/{y}{r}.png`, {
        attribution: `&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors &copy; <a href="https://carto.com/attributions">CARTO</a>`
      })
      .addTo(this._map);

    this._updateMarkers();
  }

  _updateMarkers() {
    const {initCoords, apartmentsCoords} = this.props;
    const icon = leaflet.icon({
      iconUrl: `img/map-pin.svg`,
      iconSize: [27, 39],
    });
    this._map.setView(initCoords, ZOOM);
    apartmentsCoords.map((offer) => {
      leaflet
        .marker(offer.coordinates, {icon})
        .addTo(this._map);
    });
  }
}

CitiesMap.propTypes = {
  initCoords: PropTypes.arrayOf(PropTypes.number).isRequired,
  apartmentsCoords: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    coordinates: PropTypes.arrayOf(PropTypes.number).isRequired,
  })).isRequired,
};
