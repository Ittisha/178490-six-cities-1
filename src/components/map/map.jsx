import * as React from 'react';
import PropTypes from 'prop-types';
import leaflet from 'leaflet';

export class Map extends React.PureComponent {
  render() {
    return (
      <div id="map" style={{height: 800, marginTop: 25, marginBottom: 25}} />
    );
  }

  componentDidMount() {
    this._createMap();
  }

  componentWillUnmount() {
    this.map.remove();
  }

  _createMap() {
    const {initCoords, offersCoords} = this.props;
    const icon = leaflet.icon({
      iconUrl: `img/map-pin.svg`,
      iconSize: [30, 30],
    });
    const zoom = 12;
    const map = leaflet.map(`map`, {
      center: initCoords,
      zoom,
      zoomControl: false,
      marker: true
    });

    map.setView(initCoords, zoom);

    leaflet
      .tileLayer(`https://{s}.basemaps.cartocdn.com/rastertiles/voyager/{z}/{x}/{y}{r}.png`, {
        attribution: `&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors &copy; <a href="https://carto.com/attributions">CARTO</a>`
      })
      .addTo(map);

    offersCoords.map((offer) => {
      leaflet
        .marker(offer.coordinates, {icon})
        .addTo(map);
    });
  }
}

Map.propTypes = {
  initCoords: PropTypes.arrayOf(PropTypes.number).isRequired,
  offersCoords: PropTypes.any // TODO
};
