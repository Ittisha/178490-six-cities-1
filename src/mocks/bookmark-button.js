export const MOCK_PROPS = {
  offerId: 1,
  isInBookmarks: false,
  className: ``,
  width: 15,
  height: 12,
};
