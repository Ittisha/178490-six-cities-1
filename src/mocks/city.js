export const INIT_CITY = {
  name: `Amsterdam`,
  coords: [52.38333, 4.9],
  zoom: 12,
};

export const MOCK_CITIES = [`Amsterdam`, `Paris`];

export const MOCK_CITIES_COORDS = {
  Amsterdam: [52.38333, 4.9],
  Paris: [48.864716, 2.349014],
};

export const MOCK_CITIES_ZOOM = {
  Amsterdam: 17,
  Paris: 15,
};
