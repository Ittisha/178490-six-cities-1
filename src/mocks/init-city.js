export const INIT_CITY = {
  name: `Amsterdam`,
  coords: [52.38333, 4.9],
};
