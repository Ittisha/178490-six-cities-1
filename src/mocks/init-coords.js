export const INIT_COORDS = [52.38333, 4.9];
