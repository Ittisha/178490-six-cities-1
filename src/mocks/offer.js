export const OFFER = {
  "city": {
    "name": `Cologne`,
    "location": {
      "latitude": 50.938361,
      "longitude": 6.959974,
      "zoom": 13}},
  "preview_image": `https://es31-server.appspot.com/six-cities/static/hotel/17.jpg`,
  "images": [
    `https://es31-server.appspot.com/six-cities/static/hotel/20.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/2.jpg`,
  ],
  "title": `The house among olive `,
  "is_favorite": false,
  "is_premium": false,
  "rating": 2.5,
  "type": `house`,
  "bedrooms": 2,
  "max_adults": 6,
  "price": 757,
  "goods": [
    `Coffee machine`,
    `Baby seat`,
    `Dishwasher`,
    `Fridge`,
    `Washing machine`,
    `Breakfast`,
  ],
  "host": {
    "id": 25,
    "name": `Angelina`,
    "is_pro": true,
    "avatar_url": `img/avatar-angelina.jpg`
  },
  "description": `This is a place for dreamers to reset, reflect, and create. Designed with a 'slow' pace in mind, our hope is that you enjoy every part of your stay; from making local coffee by drip in the morning, choosing the perfect record to put on as the sun sets.`,
  "location": {
    "latitude": 50.945361,
    "longitude": 6.935974,
    "zoom": 16
  },
  "id": 1
};
