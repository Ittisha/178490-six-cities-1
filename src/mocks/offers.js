export const OFFERS = [
  {
    isPremium: true,
    isInBookmarks: false,
    title: `Beautiful & luxurious apartment at great location`,
    photoUrl: `img/apartment-01.jpg`,
    price: 120,
    type: `Apartment`,
    rating: `93%`,
    id: 1,
    coordinates: [52.3909553943508, 4.85309666406198],
  },
  {
    isPremium: false,
    isInBookmarks: true,
    title: `Wood and stone place`,
    photoUrl: `img/room.jpg`,
    price: 80,
    type: `Private room`,
    rating: `80%`,
    id: 2,
    coordinates: [52.369553943508, 4.85309666406198],
  },
  {
    isPremium: false,
    isInBookmarks: false,
    title: `Canal View Prinsengracht`,
    photoUrl: `img/apartment-02.jpg`,
    price: 132,
    type: `Apartment`,
    rating: `80%`,
    id: 3,
    coordinates: [52.3909553943508, 4.929309666406198],
  },
  {
    isPremium: false,
    isInBookmarks: false,
    title: `Nice, cozy, warm big bed apartment`,
    photoUrl: `img/apartment-03.jpg`,
    price: 180,
    type: `Apartment`,
    rating: `80%`,
    id: 4,
    coordinates: [52.3809553943508, 4.939309666406198],
  }
];
