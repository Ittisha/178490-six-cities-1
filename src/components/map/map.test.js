import React from 'react';
import renderer from 'react-test-renderer';
import {Map} from './map';
import {OFFERS} from '../../mocks/offers';
import {INIT_COORDS} from '../../mocks/init-coords';

const mockCoords = OFFERS.map(({id, coordinates}) => ({id, coordinates}));

it(`Map component is rendered properly`, () => {
  const tree = renderer
    .create(<Map apartmentsCoords={mockCoords} initCoords={INIT_COORDS} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

