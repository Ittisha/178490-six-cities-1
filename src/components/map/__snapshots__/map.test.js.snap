// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map component is rendered properly 1`] = `
<div
  id="map"
  style={
    Object {
      "height": "100%",
      "minHeight": 816,
    }
  }
/>
`;
